### PR TITLE
First pass at revamping docs

### PR DESCRIPTION
**Pull request by damccorm**
Add folder dedicated for contributor docs and (a) move a bunch of info there, and (b) add some new content

My goal here is mostly to get the process started so that we can improve over time.

Part of https://github.com/GoogleCloudPlatform/DataflowTemplates/issues/1459

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,18 +24,35 @@ You generally only need to submit a CLA once, so if you've already submitted one
...
 
+## New Templates/Features
+
+This repo is generally not accepting contributions or feature requests for new
+templates or large chunks of new functionality unless a partner organization is
+willing to sponsor building/maintaining/supporting those templates in a formal
+agreement with Google.
+
+If you are part of an organization interested in this, please reach out to
+the Dataflow team directly using the [Google Issue Tracker](https://b.corp.google.com/issues/new?component=187168&template=1162867).
```
Comment (liferoad) [inline CONTRIBUTING.md:35]: is https://b.corp.google.com/issues/new?component=187168&template=1162867 accessible? I could not access it using my personal device.

File: contributor-docs/code-contributions.md [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+contribution to this repo. If you find any missing information, please update
+this guide after you find an answer!
+
+If you are a repository maintainer and need info on repo governance,
+how to review/merge code, or other info, see the
+[Maintainers Guide](./maintainers-guide.md).
+
+## Prerequisites
+
+If you are not already familiar with Dataflow, Dataflow Templates, and Apache
```
Comment (liferoad) [inline contributor-docs/code-contributions.md:13]: shall we call out flex templates should be recommended?

File: contributor-docs/code-contributions.md [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+and instructions on how to customize and/or build every template.
+
+To generate the documentation for all templates, the following command can be used:
+
+```shell
+mvn clean prepare-package \
+  -DskipTests \
+  -PtemplatesSpec
+```
+
```
Comment (liferoad) [inline contributor-docs/code-contributions.md:420]: can we add the links for python flex template builds?
https://cloud.google.com/dataflow/docs/guides/templates/using-flex-templates#python_1
https://github.com/GoogleCloudPlatform/python-docs-samples/tree/main/dataflow/flex-templates

File: contributor-docs/code-owners.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Code Owners
+
+Dataflow Templates uses [Code Owners](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners)
+to manage code review. This means that for some PRs, reviewers will be automatically assigned.
+
+The goal is for this to eventually be fully automated, it is not yet, however,
+for certain directories. If you create a PR and a reviewer is not assigned
```
Comment (liferoad) [inline contributor-docs/code-owners.md:7]: can we add the link to the ownerfile?
https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/.github/CODEOWNERS

**Review by liferoad — CHANGES_REQUESTED** [review]
Looks great! Left some minor changes.
One last question: shall we create a developer guide to show how to add a new template or integration tests, etc?

**Review by damccorm — COMMENTED** [review]
> One last question: shall we create a developer guide to show how to add a new template or integration tests, etc?

My plan is to build out docs on code contributions to have more information (and maybe split some pieces out further). I'd like to scope that to a future PR though; I'll track things with #1459

**Review by liferoad — APPROVED** [review]
Thanks.

**Pull request merged**